x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/datafeed/DatafeedConfig.java

```diff
@@ -627,74 +864,37 @@ public void setFrequency(TimeValue frequency) {
             this.frequency = frequency;
         }
 
-        public void setQuery(Map<String, Object> query) {
-            setQuery(query, true);
+        public void setQueryProvider(QueryProvider queryProvider) {
+            this.queryProvider = ExceptionsHelper.requireNonNull(queryProvider, QUERY.getPreferredName());
         }
 
-        public void setQuery(Map<String, Object> query, boolean lenient) {
-            this.query = ExceptionsHelper.requireNonNull(query, QUERY.getPreferredName());
+        // For testing only
+        public void setParsedQuery(QueryBuilder queryBuilder) {
             try {
-                QUERY_TRANSFORMER.fromMap(query);
-            } catch(Exception ex) {
-                String msg = Messages.getMessage(Messages.DATAFEED_CONFIG_QUERY_BAD_FORMAT, id);
-
-                if (ex.getCause() instanceof IllegalArgumentException) {
-                    ex = (Exception)ex.getCause();
-                }
-
-                if (lenient) {
-                    logger.warn(msg, ex);
-                } else {
-                    throw ExceptionsHelper.badRequestException(msg, ex);
-                }
+                this.queryProvider = QueryProvider.fromParsedQuery(queryBuilder);
+            } catch (IOException exception) {
+                // eat exception as it should never happen
```

Comment:
It doesn't hurt to log it as an error and just not rethrow.  If the exception never happens then nothing will ever be logged...

Same in `setParsedAggregations` below.